src/cpp/database/database.cpp

```diff
@@ -438,7 +438,11 @@ void Database::insert(
             if (participant)
             {
                 const RtpsPacketsSentSample& rtps_packets_sent = dynamic_cast<const RtpsPacketsSentSample&>(sample);
-                participant->data.rtps_packets_sent[rtps_packets_sent.remote_locator].push_back(rtps_packets_sent);
+                // Store the increment since the last report
+                participant->data.rtps_packets_sent[rtps_packets_sent.remote_locator].push_back(
+                    rtps_packets_sent - participant->data.last_reported_rtps_packets_sent_count);
+                // Update last report
+                participant->data.last_reported_rtps_packets_sent_count = rtps_packets_sent;
```

Comment:
We need to keep the `last_reported_rtps_packets_sent_count` for each locator, since we will be receiving the count for each locator separately. We cannot compute the increment for the packets to one locator with the last accumulated for another locator.

The same with `packets_lost` (and `bytes_sent` and `bytes_lost` in the future)